graylog2-server/src/main/java/org/graylog2/rest/resources/users/UsersResource.java

```diff
@@ -423,11 +421,14 @@ public void changePassword(
 
     @GET
     @Path("{username}/tokens")
-    @RequiresPermissions(RestPermissions.USERS_TOKENLIST)
     @ApiOperation("Retrieves the list of access tokens for a user")
     public TokenList listTokens(@ApiParam(name = "username", required = true)
                                 @PathParam("username") String username) {
-        final User user = _tokensCheckAndLoadUser(username);
+        final User user = _tokensLoadUser(username);
+
+        if (!isPermitted(USERS_TOKENLIST, username)) {
```

Comment:
I think we should do the check before fetching the user, so the call cannot be used to find out existing usernames.